src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditChangeDataManager.java

```diff
@@ -375,7 +376,7 @@ Entry markRead(final TimestampUTC timestamp) {
 					timestamp,
 					mIsUpvoted,
 					mIsDownvoted,
-					true,
+					PrefsUtility.pref_behaviour_mark_posts_as_read() && true,
```

Comment:
This should respect the existing value of `mIsRead`:

```java
mIsRead || PrefsUtility.pref_behaviour_mark_posts_as_read()
```

(also, `x && true` is the same as `x` :))

Reply:
Thanks, done!